Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp

```diff
@@ -1260,38 +1278,75 @@ namespace Multiplayer
         }
     }
 
-    void MultiplayerSystemComponent::EnableAutonomousControl(NetworkEntityHandle entityHandle, AzNetworking::ConnectionId connectionId)
+    void MultiplayerSystemComponent::EnableAutonomousControl(const INetworkEntityManager::EntityList& entityList, AzNetworking::ConnectionId connectionId)
     {
-        if (!entityHandle.Exists())
+        if (entityList.empty())
         {
             AZLOG_WARN("Attempting to enable autonomous control for an invalid entity");
             return;
         }
 
-        entityHandle.GetNetBindComponent()->SetOwningConnectionId(connectionId);
-        if (connectionId == InvalidConnectionId)
+        for (NetworkEntityHandle subEntity : entityList)
         {
-            entityHandle.GetNetBindComponent()->SetAllowAutonomy(true);
+            if (NetBindComponent* netbindComponent = subEntity.GetNetBindComponent())
+            {
+                netbindComponent->SetOwningConnectionId(connectionId);
+
+                if (connectionId == InvalidConnectionId)
```

Comment:
Can you add a comment explaining what's happening here? In a vacuum, Setting AllowAutonomy to true when the connection ID is invalid may seem odd to readers

Reply:
Good call! Added (now comment lives in NetworkEntityManager where autonomy is granted.